openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -0,0 +1,245 @@
+    cert_name_long = TextField()
+
+    # Grading
+    lowest_passing_grade = DecimalField(max_digits=5, decimal_places=2)
+
+    # Access parameters
+    mobile_available = BooleanField()
+    visible_to_staff_only = BooleanField()
+    _pre_requisite_courses_json = TextField()  # JSON representation of list of CourseKey strings
+
+    @staticmethod
+    def _create_from_course(course):
+        """
+        Creates a CourseOverview object from a CourseDescriptor.
+
+        Does not touch the database, simply constructs and returns an overview
+        from the given course.
+
+        Arguments:
+            course (CourseDescriptor): any course descriptor object
+
+        Returns:
+            CourseOverview: overview extracted from the given course
+        """
+        return CourseOverview(
+            id=course.id,
+            _location=course.location,
+            display_name=course.display_name,
+            display_number_with_default=course.display_number_with_default,
+            display_org_with_default=course.display_org_with_default,
+
+            start=course.start,
+            end=course.end,
+            advertised_start=course.advertised_start,
+
+            course_image_url=course_image_url(course),
+            facebook_url=course.facebook_url,
+            social_sharing_url=course.social_sharing_url,
+
+            certificates_display_behavior=course.certificates_display_behavior,
+            certificates_show_before_end=course.certificates_show_before_end,
+            has_any_active_web_certificate=(get_active_web_certificate(course) is not None),
+            cert_name_short=course.cert_name_short,
+            cert_name_long=course.cert_name_long,
+            lowest_passing_grade=course.lowest_passing_grade,
+            end_of_course_survey_url=course.end_of_course_survey_url,
+
+            mobile_available=course.mobile_available,
+            visible_to_staff_only=course.visible_to_staff_only,
+            _pre_requisite_courses_json=json.dumps(course.pre_requisite_courses)
+        )
+
+    @staticmethod
+    def get_from_id(course_id):
+        """
+        Load a CourseOverview object for a given course ID.
+
+        First, we try to load the CourseOverview from the database. If it
+        doesn't exist, we load the entire course from the modulestore, create a
+        CourseOverview object from it, and then cache it in the database for
+        future use.
+
+        Arguments:
+            course_id (CourseKey): the ID of the course overview to be loaded
+
+        Returns:
+            CourseOverview: overview of the requested course
+        """
+        course_overview = None
+        try:
+            course_overview = CourseOverview.objects.get(id=course_id)
+            # Cache hit! Just return the overview
+        except CourseOverview.DoesNotExist:
+            # Cache miss. Load entire course and create a CourseOverview from it
+            store = modulestore()
```

Comment:
Nit: I'd remove these two "Cache" comments. Someone else might confuse this querying of a DB row with actual caching in Django/memcache. And the code speaks for itself here.